src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -400,8 +410,17 @@ private void fillBuffer() {
                 buffer = null;
                 bufferIterator = null;
 
-                // TODO: heh... this should be bounded most likely...
-                fillBuffer();
+                // Why 5? Because it's less than 6.
+                if (trips >= 5) {
+                    logger.error(
+                        "Attempted to fill the buffer after an OffsetOutOfRangeException, but this was my fifth attempt so I'm bailing."
```

Comment:
i wonder if we should do it the number of partitions + 1 or something.  Or if that's just as arbitrary as 5

Reply:
There isn't a good way to determine the number of available frames are there?

Reply:
partitions? i think there is, but as your comment reads, it may not much matter because nextTuple() will be called rather frequently.

Reply:
The more I think about it, the more it probably just makes sense to have some sane limit, if 5 is what we have, 5 it is. 🤷‍♂️

Reply:
I meant frames as in the stack. Remember when we hit this we saw a stack overflow. So if we didn’t want a hard limit I was postulating we could see how many frames were left in the stack. But that seems unnecessarily complicated.